tar/bsdtar.c

```diff
@@ -187,9 +193,23 @@ static void
 bsdtar_atexit(void)
 {
 	struct bsdtar *bsdtar;
+	size_t i;
 
 	bsdtar = &bsdtar_storage;
 
+	/* Free temporary archive names (if an error occurred before export). */
+	if (bsdtar->tapenames_setup != NULL) {
+		for (i = 0; i < bsdtar->ntapes; i++)
```

Comment:
why not replace `bsdtar->ntapes` with `strlist_getsize(bsdtar->tapenames_setup)` here?  That way you won't need to adjust `bsdtar->ntapes` every time you add a string to the array.